packages/web/src/components/gcds-link/gcds-link.tsx

```diff
@@ -0,0 +1,221 @@
+  validateDisplay(newValue: string) {
+    const values = ['block', 'inline', 'inline-block'];
+
+    if (!values.includes(newValue)) {
+      this.display = 'block';
+    }
+  }
+
+  /**
+   * Link props
+   */
+
+  /**
+   * Sets the display behavior of the link
+   */
+  @Prop() display: 'block' | 'inline' | 'inline-block' = 'inline';
+
+  /**
+   * The href attribute specifies the URL of the page the link goes to
+   */
+  @Prop() href!: string;
+
+  /**
+   * The rel attribute specifies the relationship between the current document and the linked document
+   */
+  @Prop() rel: string | undefined;
+
+  /**
+   * The target attribute specifies where to open the linked document
+   */
+  @Prop() target?: string = '_self';
+
+  /**
+   * Whether the link is external or not
+   */
+  @Prop() external: false;
+
+  /**
+   * The download attribute specifies that the target (the file specified in the href attribute) will be downloaded when a user clicks on the hyperlink
+   */
+  @Prop() download: string | undefined;
+
+  /**
+   * The type specifies the media type of the linked document
+   */
+  @Prop() type: string | undefined;
+
+  /**
+   * Set additional HTML attributes not available in component properties
+   */
+  @State() inheritedAttributes: Object = {};
+
+  /**
+   * Language of rendered component
+   */
+  @State() lang: string;
+
+  /**
+   * Events
+   */
+
+  /**
+   * Emitted when the link has focus.
+   */
+  @Event() gcdsFocus!: EventEmitter<void>;
+
+  /**
+   * Emitted when the link loses focus.
+   */
+  @Event() gcdsBlur!: EventEmitter<void>;
+
+  /**
+   * Emitted when the link has been clicked.
+   */
+  @Event() gcdsClick!: EventEmitter<void>;
+
+  /*
+   * Observe lang attribute change
+   */
+  updateLang() {
+    const observer = new MutationObserver(mutations => {
+      if (mutations[0].oldValue != this.el.lang) {
+        this.lang = this.el.lang;
+      }
+    });
+    observer.observe(this.el, observerConfig);
+  }
+
+  componentWillLoad() {
+    // Validate attributes and set defaults
+    this.validateSize(this.size);
+
+    this.inheritedAttributes = inheritAttributes(this.el, this.shadowElement, [
+      'referrerpolicy',
+    ]);
+
+    // Define lang attribute
+    this.lang = assignLanguage(this.el);
+
+    this.updateLang();
+  }
+
+  render() {
+    const {
+      size,
+      lang,
+      display,
+      href,
+      rel,
+      target,
+      external,
+      download,
+      type,
+      inheritedAttributes,
+    } = this;
+
+    const attrs = {
+      href,
+      rel,
+      target,
+      download,
+      type,
+    };
+
+    const isExternal = target === '_blank' || external;
+
+    return (
+      <Host>
+        {/* eslint-disable-next-line jsx-a11y/no-static-element-interactions,jsx-a11y/click-events-have-key-events */}
+        <a
+          role="link"
+          tabIndex={0}
+          {...attrs}
+          class={`link--${size} ${display != 'block' ? `d-${display}` : ''}`}
```

Comment:
**Size**
The CSS currently only has a class for `link-small`, can we also add a class for `link-regular` and `link-inherit` to support those prop values as well?

**Block**
The CSS currently doesn't have any classes for the display prop. Can we add a class for `d-block`,  and `d-inline-block` to support those prop values? And then can we change the conditional to only add the class if it's not equal to `inline` since `inline` is supposed to be the default value?

Reply:
Yep -- good catch on the default settings for the display!